File: src/logger.lua
local function logger(tag)
    local logger = {}
    local function log(level, ...)
        print('[' .. os.date('%Y-%m-%d %H:%M:%S') .. ']', '[' .. tag .. ']', '[' .. level .. ']', ...)
    end

    -- "..." lets us toss any number of arguments to the final call to "print"
    -- (unsure if this is Lua or morse code at this point)
    function logger.debug(...)
        log('dbg', ...)
    end
    function logger.info(...)
        log('inf', ...)
    end
    function logger.warning(...)
        log('wrn', ...)
    end
    function logger.error(...)
        log('err', ...)
    end
    return logger
end

return logger

File: sharp/Logger.cs
using System;

namespace Olympus {
    public class Logger(string tag) {
        public void Debug(string msg) => log("dbg", msg);
        public void Info(string msg) => log("inf", msg);
        public void Warning(string msg) => log("wrn", msg);
        public void Error(string msg) => log("err", msg);

        private void log(string level, string msg) {
            Console.Error.WriteLine($"[{DateTime.Now:yyyy-MM-dd HH:mm:ss}] [{level}] [{tag}] {msg}");
        }
    }
}
